airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -156,57 +157,64 @@ def read(self, source_spec: ConnectorSpecification, config: TConfig, catalog: TC
         yield from self._emit_queued_messages(self.source)
 
     @staticmethod
-    def validate_connection(source_spec: ConnectorSpecification, config: Mapping[str, Any]) -> None:
+    def validate_connection(source_spec: ConnectorSpecification, config: TConfig) -> None:
         # Remove internal flags from config before validating so
         # jsonschema's additionalProperties flag won't fail the validation
         connector_config, _ = split_config(config)
         check_config_against_spec_or_exit(connector_config, source_spec)
 
     @staticmethod
-    def set_up_secret_filter(config, connection_specification: Mapping[str, Any]):
+    def set_up_secret_filter(config: TConfig, connection_specification: Mapping[str, Any]) -> None:
         # Now that we have the config, we can use it to get a list of ai airbyte_secrets
         # that we should filter in logging to avoid leaking secrets
         config_secrets = get_secrets(connection_specification, config)
         update_secrets(config_secrets)
 
     @staticmethod
-    def airbyte_message_to_string(airbyte_message: AirbyteMessage) -> str:
+    def airbyte_message_to_string(airbyte_message: AirbyteMessage) -> Any:
         return airbyte_message.json(exclude_unset=True)
 
-    def _emit_queued_messages(self, source) -> Iterable[AirbyteMessage]:
+    @classmethod
+    def extract_catalog(cls, args: List[str]) -> Optional[Any]:
```

Comment:
Where will this method be called from? I'm not sure anything should depend on the entrypoint